src/cargo/core/compiler/job_queue.rs

```diff
@@ -151,6 +157,8 @@ struct DrainState<'cfg> {
 
     // How many jobs we've finished
     finished: usize,
+    show_future_incompat_report: bool,
```

Comment:
I'm not sure I see why `show_future_incompat_report` needs to be here. It's only used in one place, where `cx` is available, so maybe it could removed and just access the `build_config` value directly?